src/select/internal.tsx

```diff
@@ -143,21 +146,21 @@ const InternalSelect = React.forwardRef(
         ref={triggerRef}
         placeholder={placeholder}
         disabled={disabled}
-        ariaLabel={ariaLabel}
         triggerVariant={triggerVariant}
         triggerProps={getTriggerProps(disabled)}
         selectedOption={selectedOption}
         isOpen={isOpen}
         inFilteringToken={__inFilteringToken}
         {...formFieldContext}
         controlId={controlId}
+        ariaLabelledby={[formFieldContext.ariaLabelledby, selectAriaLabelId].filter(label => !!label).join(' ')}
```

Comment:
We have a reusable utility for such case

https://github.com/cloudscape-design/components/blob/cff14478e987a745a26066af444337e0496a2075/src/internal/utils/strings/join-strings.ts#L4-L7

Reply:
Good point. Updated in the new revision.